crates/aptos-rest-client/src/lib.rs

```diff
@@ -501,86 +508,192 @@ impl Client {
         self.wait_for_transaction_by_hash_bcs(
             transaction.clone().committed_hash(),
             expiration_timestamp,
+            None,
+            Some(Duration::from_secs(60)),
         )
         .await
     }
 
-    pub async fn wait_for_transaction_by_hash(
+    async fn wait_for_transaction_by_hash_inner<F, Fut, T>(
         &self,
         hash: HashValue,
         expiration_timestamp_secs: u64,
-    ) -> AptosResult<Response<Transaction>> {
-        const DEFAULT_TIMEOUT: Duration = Duration::from_secs(60);
+        server_lag_timeout_after_expiry: Option<Duration>,
```

Comment:
Maybe something like `max_server_lag`? Though in addition to that we need a docstring explaining clearly what each of these timeouts do.

Reply:
For `max_server_lag`, something like:

Fullnodes generally lag some amount behind the authoritative blockchain ledger state. This field gives the node some time to update its ledger state to the point where your transaction might have expired. We recommend setting this value to x seconds.

Just off the top of my head.